src/components/Planning/PlanningGroupsTable/PlanningGroupsTableRow.tsx

```diff
@@ -3,24 +3,49 @@ import { planProjectValues } from '@/mocks/common';
 import { IconDocument, IconMenuDots } from 'hds-react/icons';
 import { FC, memo } from 'react';
 import { useNavigate } from 'react-router';
-import PlanningGroupsTableCell from './PlanningGroupsTableCell';
+import ProjectCell, { IProjectCellProps } from './ProjectCell';
 
 import { IListItem } from '@/interfaces/common';
 import { CustomTag } from '@/components/shared';
+import { isInYearRange } from '@/utils/common';
 
-/**
- * RED CELLS
- * background: 'var(--color-suomenlinna-light)'
- * borderBottom: '4px solid var(--color-suomenlinna)'
- *
- * GREEN CELLS
- * ?
- *
- * BLUE CELLS
- * ?
- */
+const createProjectCells = (project: IProject): Array<IProjectCellProps> => {
+  const getCellType = (year: number) => {
+    const { estPlanningStart, estPlanningEnd, estConstructionStart, estConstructionEnd } = project;
+    const isPlanning = isInYearRange(year, estPlanningStart, estPlanningEnd);
+    const isConstruction = isInYearRange(year, estConstructionStart, estConstructionEnd);
+
+    if (isPlanning && isConstruction) return 'planningAndConstruction';
+    if (isPlanning) return 'planning';
+    if (isConstruction) return 'construction';
+    return 'none';
+  };
+
+  const getCells = (value: string, key: string, year: number): IProjectCellProps => {
+    return {
+      value,
+      type: getCellType(year),
+      objectKey: key,
+      projectId: project.id,
+    };
+  };
+
+  const cells = [];
 
-interface IPlanningProjectsTableProps {
+  for (const [key, value] of Object.entries(project)) {
```

Comment:
return Object.entries(project).map or flatMap

Reply:
This behaviour changes in https://github.com/City-of-Helsinki/infraohjelmointi-ui/pull/38, could it be fixed there?